cogs/StaffAppsUser.py

```diff
@@ -0,0 +1,567 @@
+
+from util.EmbedBuilder import EmbedBuilder
+from util.Logging import Log
+
+
+class embeds:
+    """
+    This class creates embeds when a user is DENIED for an application.
+    See docstrings below for reasons a user can be DENIED.
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    def min_reqs(
+        self,
+        msg_count: int,
+        join_time: datetime,
+        created_at: datetime,
+    ) -> discord.Embed:
+        """
+        User does not meet the minimum requirements to apply for staff.
+        """
+        time_since_creation = datetime.now(tz=created_at.tzinfo) - created_at
+        time_since_join = datetime.now(tz=join_time.tzinfo) - join_time
+
+        fields = [
+            [
+                "Created",
+                f"""{created_at.strftime('%B %d, %Y')}
+                {precisedelta(time_since_creation, minimum_unit='days', format="%.0F")} ago""",
+                True,
+            ],
+            [
+                "Joined",
+                f"""{join_time.strftime('%B %d, %Y')}
+                {precisedelta(time_since_join, minimum_unit='days', format="%.0F")} ago""",
+                True,
+            ],
+            ["Messages", f"{msg_count} messages", True],
+        ]
+
+        wait_time = max(
+            timedelta(weeks=52) - time_since_creation,
+            timedelta(days=30) - time_since_join,
+        )
+        desc = """Unfortunately, you do not meet the basic requirements in order to apply for staff.
+            Your account must be at least 1 year old, you must have been a member of the server for at least 30 days and you need to have at least 500 messages sent total in any channel. (sent after November 20th 2022)"""
+        if wait_time > timedelta(0):
+            desc += f"""\n\nYou will be eligible to apply for staff in **{precisedelta(wait_time, minimum_unit='days', format="%.0F")}**."""
+        if msg_count < 500:
+            desc += f"""\n\nYou need to send **{(left := 500 - msg_count)}** more message{"" if left == 1 else "s"}."""
+        return EmbedBuilder(
+            title="Missing Requirements",
+            description=desc,
+            fields=fields,
+            color=0xFF0000,  # RED
+        ).build()
+
+    def marked_spam(self) -> discord.Embed:
+        """
+        User is marked spam by staff for a previous offence.
+        """
+        return EmbedBuilder(
+            title="You cannot apply for staff!",
+            description="Unfortunately, you cannot apply for staff at this time. If you believe this is a mistake, please contact a staff member.",
+            color=0xFF0000,  # RED
+        ).build()
+
+    def cooldown(self, cooldown: int) -> discord.Embed:
+        """
+        User has recently applied for staff and was denied.
+        A cooldown is in place to prevent a user from immediatly re-applying.
+        """
+        curtime = datetime.now()
+        cooldown_time = datetime.fromtimestamp(cooldown)
```

Comment:
Please add `tz` argument

Reply:
datetime.fromtimestamp() takes an EPOCH timestamp as argument, EPOCH timestamps are the time in seconds elapsed since 00:00:00 UTC on 1 January 1970. This object is already timezone-aware at the time of creation.